salt/modules/win_snmp.py

```diff
@@ -33,17 +42,33 @@ def __virtual__():
     '''
     Only works on Windows systems.
     '''
-    if salt.utils.is_windows():
-        return __virtualname__
-    return False
+    if not salt.utils.is_windows():
+        return False, 'Module win_snmp: Requires Windows'
+
+    if not __salt__['reg.read_value'](_HKEY, _SNMP_KEY)['success']:
+        return False, 'Module win_snmp: SNMP not installed'
+
+    return __virtualname__
+
+
+def _to_unicode(instr):
+    '''
+    Converts from current users character encoding to unicode.
+    When instr has a value of None, the return value of the function
+    will also be None.
+    '''
+    if instr is None or isinstance(instr, six.text_type):
+        return instr
+    else:
+        return six.text_type(instr, 'utf8')
```

Comment:
Why not `__salt_system_encoding__` since it now handles(should at least) windows specifically?